rust/main/agents/scraper/src/agent.rs

```diff
@@ -114,19 +88,47 @@ impl BaseAgent for Scraper {
         tasks.push(server_task);
 
         for (domain, scraper) in self.scrapers.iter() {
-            tasks.push(self.scrape(*domain).await);
+            match self.scrape(*domain).await {
+                Ok(scraper_task) => {
+                    tasks.push(scraper_task);
+                }
+                Err(err) => {
+                    tracing::error!(?err, ?scraper.domain, "Failed to scrape domain");
+                    self.chain_metrics
+                        .set_critical_error(scraper.domain.name(), true);
+                    continue;
+                }
+            }
 
-            let chain_conf = self.settings.chain_setup(&scraper.domain).unwrap();
-            let metrics_updater = MetricsUpdater::new(
+            let chain_conf = match self.settings.chain_setup(&scraper.domain) {
+                Ok(s) => s,
+                Err(err) => {
+                    tracing::error!(?err, ?scraper.domain, "Failed to get chain config");
+                    self.chain_metrics
+                        .set_critical_error(scraper.domain.name(), true);
+                    continue;
+                }
+            };
+
+            match MetricsUpdater::new(
                 chain_conf,
                 self.core_metrics.clone(),
                 self.agent_metrics.clone(),
                 self.chain_metrics.clone(),
                 Self::AGENT_NAME.to_string(),
             )
             .await
-            .unwrap();
-            tasks.push(metrics_updater.spawn());
+            {
+                Ok(metrics_updater) => {
+                    tasks.push(metrics_updater.spawn());
+                }
+                Err(err) => {
+                    tracing::error!(?err, ?scraper.domain, "Failed to build metrics updater");
+                    self.chain_metrics
+                        .set_critical_error(scraper.domain.name(), true);
+                    continue;
+                }
+            }
         }
         if let Err(err) = try_join_all(tasks).await {
```

Comment:
I have a concern regarding possible situation when we have scraper tasks running for a domain while getting `ChainConf` and `MetricsUpdater` fails. We shall have a scraper running without proper observability.

Reply:
So would you recommend rearranging the order like so?
```
build chain_conf
build metrics_updater
build scraper task
```

Reply:
Yes, it can be a solution. I would also build all tasks first and spawn them only if we can build all of them. To be specific, I propose the following:
1. build `chain_conf`
2. build `metric_updater` but don't spawn it
3. build scraper task (and all internal tasks, effectively it is just invoke `scrape` method)
4. spawn `metric_updater` task